### PR TITLE
Apply fixes suggested by scan-build static analyzer and pgindent

### PR DESCRIPTION
**Pull request by cevian**

### CODE REVIEW

File: src/planner_utils.c [hunk 1]
```diff
@@ -7,7 +7,8 @@
 static void plantree_walker(Plan **plan, void (*walker) (Plan **, void *), void *ctx);
 
 static inline void
-plantree_walk_subplans(List *plans, void (*walker) (Plan **, void *), void *ctx)
+plantree_walk_subplans(List *plans,
```
Comment (erimatnor) [inline src/planner_utils.c]: What's going on here? Is this some diff in how `pgindent` works?

Comment (cevian) [reply src/planner_utils.c]: so my pgindent indents this function here if I don't add this newline. Yours doesn't?

File: src/dimension_slice.c [hunk 2]
```diff
@@ -25,7 +25,6 @@ dimension_slice_from_form_data(Form_dimension_slice fd)
 {
 	DimensionSlice *slice = dimension_slice_alloc();
 
-	slice = palloc0(sizeof(DimensionSlice));
```
Comment (cevian) [inline src/dimension_slice.c]: This is the only actual fix I think.

**Comment by cevian** [comment]
@erimatnor yeah there was only one actual fix (suggested above). The rest the static analyzer couldn't figure out those values would not be NULL (without the Assert). I figured if the analyzer couldn't figure it out then that's a good indication we should add an Assert.

**Pull request merged**